synapse/rest/client/v2_alpha/sync.py

```diff
@@ -115,6 +115,8 @@ def on_GET(self, request):
             )
         )
 
+        request_key = (user, timeout, since, filter_id, full_state)
```

Comment:
I'm nervous about hard coding the query parameters that we care about, it feels like it would be too easy to forget to add them to the list. The failure mode of that is fairly terrible, too.